aiida_mlip/workflows/ht_workgraph.py

```diff
@@ -55,94 +54,35 @@ def build_ht_calc(
-        calc_inputs[input_struct_key] = structure
-        calc_task = wg.add_task(
-            calc,
-            name=f"calc_{file.stem}",
-            **calc_inputs,
-        )
-
-        wg.update_ctx(
-            {f"structs.{file.stem}": getattr(calc_task.outputs, final_struct_key)}
-        )
-
-    wg.outputs.final_structures = wg.ctx.structs
-
-    if structure is None:
-        raise FileNotFoundError(
-            f"{folder} is empty or has no readable structure files."
-        )
-
-    return wg
-
-
-def get_ht_workgraph(
-    calc: CalcJob | Callable | WorkChain | WorkGraph,
-    folder: Path | str | Str,
-    calc_inputs: dict,
-    input_struct_key: str = "struct",
-    final_struct_key: str = "final_structure",
-    recursive: bool = True,
-    max_number_jobs: int = 10,
-) -> WorkGraph:
-    """
-    Get WorkGraph to carry out calculation on all structures in a directory.
-
-    Parameters
-    ----------
-    calc : Union[CalcJob, Callable, WorkChain, WorkGraph]
-        Calculation to be performed on all structures.
-    folder : Union[Path, str, Str]
-        Path to the folder containing input structure files.
-    calc_inputs : dict
-        Dictionary of inputs, shared by all the calculations. Must not contain
-        `struct_key`.
-    input_struct_key : str
-        Keyword for input structure for `calc`. Default is "struct".
-    final_struct_key : str
-        Key for final structure output from `calc`. Default is "final_structure".
-    recursive : bool
-        Whether to search `folder` recursively. Default is True.
-    max_number_jobs : int
-        Max number of subprocesses running within the WorkGraph. Default is 10.
-
-    Returns
-    -------
-    WorkGraph
-        The workgraph ready to be submitted.
-    """
-    wg = WorkGraph("ht_calculation")
-    wg.add_input("workgraph.any", "calc_inputs")
-
-    wg.add_task(
-        build_ht_calc,
-        name="ht_calc",
-        calc=calc,
-        folder=folder,
-        calc_inputs=wg.inputs.calc_inputs,
-        input_struct_key=input_struct_key,
-        final_struct_key=final_struct_key,
-        recursive=recursive,
-    )
-
-    wg.inputs.calc_inputs = calc_inputs
-    wg.outputs.final_structures = wg.tasks.ht_calc.outputs.final_structures
-    wg.max_number_jobs = max_number_jobs
-
-    return wg
+    with WorkGraph("ht_calculation") as wg:
+        pattern = "**/*" if recursive else "*"
+        for file in filter(Path.is_file, folder.glob(pattern)):
+            try:
+                read(file)
+            except Exception:
+                continue
+            structure = load_structure(file).store()
+            calc_inputs[input_struct_key] = structure
+
+            calc_task = wg.add_task(
+                calc,
+                name=f"calc_{file.stem}",
+                **calc_inputs,
+            )
+            structs[file.stem] = getattr(calc_task.outputs, final_struct_key)
+
+        wg.outputs.final_structure = structs
```

Comment:
It's pretty minor, but why is `final_structure` singular now?

In general we want variable names to give context, but particularly for an output designed specifically to be interacted with by users, the distinction between `structure` and `structures` is useful.